src/view.cpp

```diff
@@ -335,20 +335,28 @@ void View::open_in_pager(const std::string& filename)
 	pop_current_formaction();
 }
 
-nonstd::optional<std::uint8_t> View::open_in_browser(const std::string& url)
+nonstd::optional<std::uint8_t> View::open_in_browser(const std::string& url,
+	const std::string& feedurl)
 {
 	std::string cmdline;
 	const std::string browser = cfg->get_configvalue("browser");
 	const std::string escaped_url = "'" + utils::replace_all(url, "'", "%27") + "'";
-	if (browser.find("%u") != std::string::npos) {
-		cmdline = utils::replace_all(browser, "%u", escaped_url);
+	const std::string escaped_feedurl = "'" + utils::replace_all(feedurl, "'",
+			"%27") + "'";
+
+	FmtStrFormatter fmt;
```

Comment:
I think using `FmtStrFormatter` for this is a bad idea, because the formatter also expands various format sequences like `%>x`, `%?cond?then&else?`, etc.

Unfortunately it looks like you can't just `replace_all(replace_all(browser, "%u", url), "%F", feedurl)` either — this might wreck the result of %u if it contains %F. It appears that you'll have to employ some iterators and walk over the string, replacing each placeholder in turn.

Reply:
I introduced an overload for `utils::replace_all()` which supports multiple pairs of from/to pairs: https://github.com/newsboat/newsboat/pull/1314/commits/4931232ee5eb78ef6fae86bceccf00445e9539d9